lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -53,7 +53,8 @@ sub register {
 }
 
 sub _child {
-  my ($c, $args, $stdin, $stdout, $stderr) = @_;
+  my ($defaults, $c, $args, $stdin, $stdout, $stderr) = @_;
+  my $pre_exec = $args->{pre_exec} || $defaults->{pre_exec};
```

Comment:
I don't like `pre_exec`, since the code that is started in the child can be both a code block or an application.